### PR TITLE
Refactor FXIOS-6867 [v119] Move CFR to component library

### PR DESCRIPTION
**Pull request by lmarceau**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-6867)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/15300)

## :bulb: Description
- Unrelated, but noticed an unused variable in BVC so removed it.
- Move the CFR to the component library
- Add a sample app example.
Simulator Screenshot - iPhone 14 Pro - 2023-09-14 at 11 18 12

## :pencil: Checklist
You have to check all boxes before merging
- [X] Filled in the above information (tickets numbers and description of your work)
- [X] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [X] Wrote unit tests and/or ensured the tests suite is passing
- [X] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [X] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: Client.xcodeproj/xcshareddata/xcschemes/Fennec.xcscheme [hunk 1]
```diff
@@ -140,8 +140,8 @@
    </BuildAction>
    <TestAction
       buildConfiguration = "Fennec"
-      selectedDebuggerIdentifier = ""
-      selectedLauncherIdentifier = "Xcode.IDEFoundation.Launcher.PosixSpawn"
+      selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
```
Comment (yoanarios) [inline Client.xcodeproj/xcshareddata/xcschemes/Fennec.xcscheme:143]: Curious wha this change is and if it was intentional

**Review by yoanarios — APPROVED** [review]
LGTM just a small question

File: Client.xcodeproj/xcshareddata/xcschemes/Fennec.xcscheme (hunk 1 above)
Comment (lmarceau) [reply Client.xcodeproj/xcshareddata/xcschemes/Fennec.xcscheme:143]: This change is to be able to debug tests, I got annoyed that I couldn't put breakpoints in my unit tests.
 
It's this option here to debug executable, not sure why it was disabled:
Screenshot 2023-09-15 at 11 24 57 AM

Comment (yoanarios) [reply Client.xcodeproj/xcshareddata/xcschemes/Fennec.xcscheme:143]: Good to know I had the same problem when I was doing the test for Redux

**Comment by lmarceau** [comment]
Merging since it's failing on main/other PRs as well for this UI test

**Pull request merged**